src/mame/drivers/swtpc8212.cpp

```diff
@@ -1,19 +1,57 @@
 // license:BSD-3-Clause
-// copyright-holders:AJR
+// copyright-holders:AJR,68bit
 /****************************************************************************
 
-    Skeleton driver for Southwest Technical Products video terminal.
+    Driver for Southwest Technical Products video terminal.
+
+The 8212 terminal appears similar in design to the CT-82 terminal for which
+there is some documentation and a user guide at:
+http://www.swtpc.com/mholley/CT_82/CT82_Index.htm
+
+The 8212 has three CRT controller configurations:
+ 1. 82 x 20 characters, with a height of 14 scan lines.
+ 2. 82 x 24 characters, with a height of 12 scan lines.
+ 3. 92 x 22 characters, with a height of 12 scan lines.
+
+There are two character generators:
+1. The MCM66750 with 7x9 sized characters with some descending by 3 lines
+giving an effective 7x12 sized character set.
+2. An EPROM defining 8x16 sized characters for which only 12 scan lines are
+used, and this is used for the graphics mode.
+
+These appear to map onto the CT-82 screen formats options as follows:
+Format I:   82 by 20, MCM66750 character set. Ctrl-\ crtl-Q
+Format II:  82 by 24, MCM66750 character set. Ctrl-\ crtl-R
+Format III: 82 by 20, EPROM character set.    Ctrl-\ crtl-S
+Format IV:  82 by 24, EPROM character set.    Ctrl-\ crtl-T
+Graphics:   92 by 22, EPROM character set.    Ctrl-] crtl-V
+
+The terminal includes a parallel printer output and Ctrl-] Ctrl-K enables
+printer pass-through and Ctrl-] Ctrl-G disables pass-through.
+
+The terminal appears to include light pen support, and the positions is read
+via Ctrl-] Ctrl-B, TODO.
+
+TODO A generic 'beep' is used. Might want to compare this with the actual 8212
+to better emulate the sound.
+
+TODO A generic keyboard is used. Might need some work to get a better keyboard
+emulation. There also appear to be separate 'cursor' key inputs, with a
+separate strobe, to emulate.
 
 ****************************************************************************/
 
 #include "emu.h"
-//#include "bus/rs232/rs232.h"
 #include "cpu/m6800/m6800.h"
 #include "machine/6821pia.h"
 #include "machine/input_merger.h"
 #include "machine/ins8250.h"
 #include "video/mc6845.h"
 #include "screen.h"
+#include "speaker.h"
+#include "imagedev/printer.h"
+#include "machine/terminal.h"
+#include "bus/rs232/rs232.h"
```

Comment:
Please keep the device includes above the ones from src/emu, and keep them sorted.